autoPyTorch/datasets/base_dataset.py

```diff
@@ -106,14 +108,16 @@ def __init__(
             val_transforms (Optional[torchvision.transforms.Compose]):
                 Additional Transforms to be applied to the validation/test data
         """
-        self.dataset_name = dataset_name if dataset_name is not None \
-            else hash_array_or_matrix(train_tensors[0])
+        self.dataset_name = dataset_name
+
+        if self.dataset_name == "":
+            self.dataset_name = str(uuid.uuid1(clock_seq=os.getpid()))
```

Comment:
Then if you think that it should not be required, then maybe do it as:
```
self.dataset_name = dataset_name if dataset_name is not None else str(
    uuid.uuid1(clock_seq=os.getpid())
)
```

What do you think?

Reply:
Yeah, actually I am also thinking about it this way.
Probably, it will be better.

Reply:
Ah, wait. But when I think about the case where we would like to use totally new datasets which do not have any name, probably we would like to choose our own name. In this sense, it is better to get back to `Optional`.